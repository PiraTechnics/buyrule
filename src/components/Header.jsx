import { Link } from "react-router-dom";
import Brand from "./Brand";
import { Navbar } from "flowbite-react";
import Cart from "./Cart";

const Header = () => {
	return (
		<div className="w-screen bg-sky-600">
			<Navbar className="bg-sky-600 py-2 mx-auto max-w-7xl ps-8 pe-12 relative flex justify-between">
				<Brand />
				<Navbar.Toggle className="sm:hidden mx-6 bg-white hover:bg-slate-300 hover:text-slate-700 focus:ring-none hover:border-amber-300" />
				<Navbar.Collapse className="sm:block sm:w-auto">
					<div
						id="navbar-menu"
						className="flex flex-row max-sm:flex-col max-sm:text-center gap-4 max-sm:gap-0 max-sm:me-4 max-sm:border-t text-lg font-hylia"
					>
						<Link
							to="/"
							className="text-white hover:text-yellow-200 max-sm:hover:bg-sky-800 py-2"
						>
							Home
						</Link>
						<Link
							to="/shop"
							className="text-white hover:text-yellow-200 max-sm:hover:bg-sky-800 py-2"
						>
							Shopping
						</Link>
						{/* drawer init and toggle */}
						<div className="text-center">
							<button
								className="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
								type="button"
								data-drawer-target="drawer-right-example"
								data-drawer-show="drawer-right-example"
								data-drawer-placement="right"
								aria-controls="drawer-right-example"
							>
								Cart
							</button>
						</div>
					</div>
				</Navbar.Collapse>
				<Cart />
			</Navbar>
		</div>
	);
};

export default Header;
