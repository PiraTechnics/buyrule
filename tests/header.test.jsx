import { render, screen } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import Header from "/src/components/Header";
import { BrowserRouter } from "react-router-dom";
//import userEvent from "@testing-library/user-event";

beforeEach(() => {
  render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );
});

describe("Header Component", () => {
  it("contains a navbar", () => {
    //screen.debug();
    expect(screen.getByRole("navigation")).toBeInTheDocument();
  });

  it("has nav links to home and shopping", () => {
    expect(screen.getAllByRole("link").length).toBe(2);
  });

  it("has a 'home' link", () => {
    expect(screen.getAllByRole("link")[0].textContent).toMatch(/home/i);
  });

  it("has a 'shop' link", () => {
    expect(screen.getAllByRole("link")[1].textContent).toMatch(/shopping/i);
  });

  it("hasn't changed unexpectedly", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});

/* describe("Navigation Links", () => {
  const user = userEvent.setup();

  it("takes a user to the home page when 'home' is clicked", async () => {

    const homeLink = screen.getByRole("link", { name: "Shopping" });
    await user.click(homeLink);

    screen.debug();
  });
}); */
