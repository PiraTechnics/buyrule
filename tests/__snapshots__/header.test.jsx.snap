// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > hasn't changed unexpectedly 1`] = `
<div>
  <div
    class="w-screen bg-sky-600"
  >
    <nav
      class="px-2 dark:border-gray-700 dark:bg-gray-800 sm:px-4 bg-sky-600 py-2 mx-auto max-w-7xl ps-8 pe-12 relative flex justify-between"
    >
      <div
        class="mx-auto flex flex-wrap items-center justify-between container"
      >
        <div
          class="text-4xl text-white tracking-logo font-hylia"
        >
          <div
            class="-mb-2 flex flex-row"
          >
            BUY
            <img
              alt="rupee icon"
              class="w-8 pt-2 ms-1"
              src="/src/assets/rupee-logo-tilted.svg"
            />
          </div>
          <div
            class=""
          >
            RULE
          </div>
        </div>
        <button
          class="inline-flex items-center rounded-lg p-2 text-sm text-gray-500 focus:outline-none focus:ring-2 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600 md:hidden sm:hidden mx-6 bg-white hover:bg-slate-300 hover:text-slate-700 focus:ring-none hover:border-amber-300"
          data-testid="flowbite-navbar-toggle"
        >
          <span
            class="sr-only"
          >
            Open main menu
          </span>
          <svg
            aria-hidden="true"
            class="h-6 w-6 shrink-0"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </button>
        <div
          class="w-full md:block md:w-auto hidden sm:block sm:w-auto"
          data-testid="flowbite-navbar-collapse"
        >
          <ul
            class="mt-4 flex flex-col md:mt-0 md:flex-row md:space-x-8 md:text-sm md:font-medium"
          >
            <div
              class="flex flex-row max-sm:flex-col max-sm:text-center gap-4 max-sm:gap-0 max-sm:me-4 max-sm:border-t text-lg font-hylia"
              id="navbar-menu"
            >
              <a
                class="text-white hover:text-yellow-200 max-sm:hover:bg-sky-800 py-2"
                href="/"
              >
                Home
              </a>
              <a
                class="text-white hover:text-yellow-200 max-sm:hover:bg-sky-800 py-2"
                href="/shop"
              >
                Shopping
              </a>
              <div
                class="text-center"
              >
                <button
                  aria-controls="shopping-cart-sidebar"
                  class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
                  data-drawer-placement="right"
                  data-drawer-show="shopping-cart-sidebar"
                  data-drawer-target="shopping-cart-sidebar"
                  type="button"
                >
                  Cart
                </button>
              </div>
            </div>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
